packages/core/src/routes/session.ts

```diff
@@ -581,12 +582,18 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
   router.get('/sign-in-settings', async (ctx, next) => {
     const signInExperience = await findDefaultSignInExperience();
     const connectorInstances = await getConnectorInstances();
-    const instanceMap = new Map(
-      connectorInstances.map((instance) => [instance.connector.id, instance])
-    );
-    const socialConnectors = signInExperience.socialSignInConnectorIds.map((id) => {
-      return { ...instanceMap.get(id)?.metadata, id };
-    });
+    const socialConnectors = signInExperience.socialSignInConnectorTargets.reduce<
+      Array<ConnectorMetadata & { id: string }>
+    >((previous, connectorTarget) => {
+      const connectors = connectorInstances.filter(
+        ({ metadata: { target } }) => target === connectorTarget
+      );
+
+      return [
+        ...previous,
+        ...connectors.map(({ metadata, connector: { id } }) => ({ ...metadata, id })),
+      ];
```

Comment:
nit: better add UTs for multiple connectors with the same `target` in one group.

Reply:
good point, updated